internal/metadataproviders/aws/ec2/metadata.go

```diff
@@ -22,9 +23,14 @@ type metadataClient struct {
 
 var _ Provider = (*metadataClient)(nil)
 
+// New Provider returns a new AWS EC2 metadata provider. The provider is established
+// with a custom logger to suppress any SDK logs, in order to work around an issue in the AWS SDK Go
+// (https://github.com/aws/aws-sdk-go/issues/5116), which causes unwanted output to appear on
+// the stdout. This change can be reverted once the issue is resolved upstream and a new version
```

Comment:
Can we add a tracking issue for this?

Reply:
Hope this suffices https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30139